src/alfasim_sdk/_internal/alfacase/case_description_attributes.py

```diff
@@ -341,6 +341,19 @@ def attrib_instance(type_: type) -> attr._make._CountingAttr:
     )
 
 
+def attrib_optional_instance(type_: type) -> attr._make._CountingAttr:
+    """
+    Create a new optional attr attribute with validator for the given type_.
+    """
+    metadata = {"type": "instance", "class_": Optional[type_]}
+    return attr.ib(
+        default=None,
+        validator=optional(instance_of(type_)),
+        type=Optional[type_],
+        metadata=metadata,
+    )
```

Comment:
Also.
For example `attrib_curve` we got an `is_optional` flag instead of having an `attrib_optional_curve`:

https://github.com/ESSS/alfasim-sdk/blob/08b36bd85eea6a93187fed0b914dcb7f0abd4498/src/alfasim_sdk/_internal/alfacase/case_description_attributes.py#L273-L328

Maybe adding that flag to `attrib_instance` will make maintenance easier if we need to change the metadata contents.